koku/masu/processor/parquet/parquet_report_processor.py

```diff
@@ -239,6 +241,14 @@ def post_processor(self):
             post_processor = ocp_post_processor
         return post_processor
 
+    @property
+    def col_checker(self):
+        """Customer column checker based on provider type."""
+        col_checker = None
+        if self.provider_type in [Provider.PROVIDER_AWS, Provider.PROVIDER_AWS_LOCAL]:
+            col_checker = check_aws_custom_columns
+        return col_checker
```

Comment:
This interface seems odd, and it hard codes specific behavior in the base class. Passing the checker into the object constructor and raising `NotImplementedError("No custom column checker defined")` creates better boundaries between objects and would more clearly show an incorrect use of the API.

Currently, if the caller forgets to test before calling `self.col_checker()` the caller will see `TypeError: 'NoneType' object is not callable` which isn't a clear indication of the problem.

Reply:
That's ok. Let's not worry about it.